### DISCUSSION TITLE
Theme Wallpaper Settings/Customization (Desktop Fit and Timings)

### DISCUSSION THREAD

**Discussion by 47prince** [t1m0thyj/WinDynamicDesktop#623] [Q&A] [answered]
Whenever the wallpaper transitions, it switches the desktop fit to 'fill' and some themes have ultrawide dimensions (different to mine) so the wallpaper zooms in too much. Is there a way to change the default setting so that I don't have to do it in Windows settings for each time transition?

Moreover, for themes that have multiple wallpapers (for times in the day and night), is it possible to change the timing for specific ones to be earlier/later? (There is a specific wallpaper I like but it switches too late in the night for me to enjoy).

I know we can create scripts but I am not sure if that's what is needed to make these changes, or if there's an easier way.

**Comment by t1m0thyj** [comment, marked as answer]
> Whenever the wallpaper transitions, it switches the desktop fit to 'fill' and some themes have ultrawide dimensions (different to mine) so the wallpaper zooms in too much. Is there a way to change the default setting so that I don't have to do it in Windows settings for each time transition?

If you are using the Microsoft Store version of WDD, this is a known issue - see the workaround [here](https://github.com/t1m0thyj/WinDynamicDesktop/wiki/Known-issues#wallpaper-fit-not-saved-with-multiple-monitors)

> Moreover, for themes that have multiple wallpapers (for times in the day and night), is it possible to change the timing for specific ones to be earlier/later? (There is a specific wallpaper I like but it switches too late in the night for me to enjoy).

You can configure a custom sunrise/sunset time in WDD to be earlier/later than the default times for your location. There isn't a way to change the timing for specific themes without editing the image sequence in their theme.json files.

**Reply by 47prince** [reply]
Thanks. Currently, migrating to the GitHub version might be the most convenient solution.

> There isn't a way to change the timing for specific themes without editing the image sequence in their theme.json files.

In the theme.json, there only seems to be "nightImageList" (for example) and the sequence for night mode images, but no time setting for each image. Can one change the timing for the specific theme mode images?

**Reply by t1m0thyj** [reply]
WDD doesn't support changing the timing for specific images. For the case you mentioned earlier:
> There is a specific wallpaper I like but it switches too late in the night for me to enjoy

There may be ways you can make it switch earlier, since images can be repeated in the sequence as often as desired. For example if the specific wallpaper you like is the first item in `nightImageList`, you could make it appear earlier by adding it to the end of `sunsetImageList` (or `dayImageList` for a 2-segment theme).
